src/main/java/org/motechproject/mots/service/ModuleAssignmentService.java

```diff
@@ -97,9 +107,10 @@ public void assignModules(AssignedModules assignedModules) {
 
     Set<Module> modulesToAdd = getModulesToAdd(oldModules, newModules);
     Set<Module> modulesToRemove = getModulesToRemove(oldModules, newModules);
+    validateModulesToUnassign(assignedModulesChw, modulesToRemove);
+    moduleProgressService.removeModuleProgresses(assignedModulesChw, modulesToRemove);
```

Comment:
move removeModuleProgresses at the end of this method next to createModuleProgresses